.github/workflows/pythonpackage.yml

```diff
@@ -0,0 +1,92 @@
+name: Python package
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        #  ** Update flake8 literals if change os or python-versions
+        os: [ubuntu-latest, windows-latest] # , macos-latest]
+        python-version:  [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Get pip cache dir
+      id: pip-cache
+      run: |
+        python -m pip install -U pip # to ensure version > 20 to have cache dir
+        echo "::set-output name=dir::$(pip cache dir)"
+    - name: pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: py${{ matrix.python-version }}-${{ matrix.os }}-pip
+    - name: Initial test environment setup
+      run: |
+        python -m pip install wheel 
+        python -m pip install pytest pytest-cov
+    - name: Flake8 checks
```

Comment:
Would it make more sense to run the flake8 checks as a separate workflow? I'd imagine failing a build early because of a style issue would get annoying very quickly. Except at the end, I'm usually more interested in the unit tests.